### PR TITLE
DOMA-1731 move map container when interacting with the last section

### PR DESCRIPTION
**Pull request by sitozzz**
#### Solution: in some cases we add extra space to the right side of container
image

### CODE REVIEW

**Review by dkoviazin — APPROVED** [review]

File: apps/condo/domains/property/components/panels/Builder/BuildingPanelEdit.tsx [hunk 1]
```diff
@@ -449,16 +449,38 @@ const CHESS_SCROLL_CONTAINER_STYLE: React.CSSProperties = {
...
+            if (Builder.editMode === 'addSection') {
+                container.current.style.paddingRight = SCROLL_CONTAINER_EDIT_PADDING
+            } else if (Builder.editMode === 'editSection') {
+                // When user select last section we actually need to move container to the left side of screen
+                const needToAddPadding = get(Builder.getSelectedSection(), 'index') === Builder.lastSectionIndex
+                needToAddPadding && (container.current.style.paddingRight = SCROLL_CONTAINER_EDIT_PADDING)
+            } else if (Builder.editMode === 'addUnit' || Builder.editMode === 'editUnit') {
+                // Last case when user want to add or edit unit only at the last section
+                const needToAddPadding = get(Builder.getSelectedUnit(), 'sectionIndex') === Builder.lastSectionIndex
+                needToAddPadding && (container.current.style.paddingRight = SCROLL_CONTAINER_EDIT_PADDING)
```
Comment (akarjakin) [inline apps/condo/domains/property/components/panels/Builder/BuildingPanelEdit.tsx:475]: `if (needToAddPadding) container.current.style.paddingRight = SCROLL_CONTAINER_EDIT_PADDING`

File: apps/condo/domains/property/components/panels/Builder/BuildingPanelEdit.tsx [hunk 2]
```diff
@@ -449,16 +449,38 @@ const CHESS_SCROLL_CONTAINER_STYLE: React.CSSProperties = {
...
+            : 0
+        const isNeedToMoveContainer = Builder.editMode !== null && !Builder.isEmpty && childTotalWidth > MENU_COVER_MAP_WIDTH
+        if (isNeedToMoveContainer) {
+            // Always if modal for new section was opened we need to move container to the left
+            if (Builder.editMode === 'addSection') {
+                container.current.style.paddingRight = SCROLL_CONTAINER_EDIT_PADDING
+            } else if (Builder.editMode === 'editSection') {
+                // When user select last section we actually need to move container to the left side of screen
+                const needToAddPadding = get(Builder.getSelectedSection(), 'index') === Builder.lastSectionIndex
+                needToAddPadding && (container.current.style.paddingRight = SCROLL_CONTAINER_EDIT_PADDING)
```
Comment (akarjakin) [inline apps/condo/domains/property/components/panels/Builder/BuildingPanelEdit.tsx:471]: `if (needToAddPadding) container.current.style.paddingRight = SCROLL_CONTAINER_EDIT_PADDING`

File: apps/condo/domains/property/components/panels/Builder/BuildingPanelEdit.tsx [hunk 3]
```diff
@@ -449,16 +449,38 @@ const CHESS_SCROLL_CONTAINER_STYLE: React.CSSProperties = {
...
+            ? Array.from(container.current.children).reduce((total, element) => total + element.clientWidth, 0)
+            : 0
+        const isNeedToMoveContainer = Builder.editMode !== null && !Builder.isEmpty && childTotalWidth > MENU_COVER_MAP_WIDTH
+        if (isNeedToMoveContainer) {
+            // Always if modal for new section was opened we need to move container to the left
+            if (Builder.editMode === 'addSection') {
+                container.current.style.paddingRight = SCROLL_CONTAINER_EDIT_PADDING
+            } else if (Builder.editMode === 'editSection') {
+                // When user select last section we actually need to move container to the left side of screen
+                const needToAddPadding = get(Builder.getSelectedSection(), 'index') === Builder.lastSectionIndex
```
Comment (akarjakin) [inline apps/condo/domains/property/components/panels/Builder/BuildingPanelEdit.tsx:470]: `shouldAddPadding`

File: apps/condo/domains/property/components/panels/Builder/BuildingPanelEdit.tsx [hunk 4]
```diff
@@ -449,16 +449,38 @@ const CHESS_SCROLL_CONTAINER_STYLE: React.CSSProperties = {
...
     useEffect(() => {
-        if (container.current) {
-            if (Builder.previewSectionId) {
-                const { scrollWidth, clientWidth, scrollHeight, clientHeight } = container.current
-                container.current.scrollTo(scrollWidth - clientWidth, scrollHeight - clientHeight)
+        const childTotalWidth = container.current !== null
+            ? Array.from(container.current.children).reduce((total, element) => total + element.clientWidth, 0)
+            : 0
+        const isNeedToMoveContainer = Builder.editMode !== null && !Builder.isEmpty && childTotalWidth > MENU_COVER_MAP_WIDTH
+        if (isNeedToMoveContainer) {
```
Comment (akarjakin) [inline apps/condo/domains/property/components/panels/Builder/BuildingPanelEdit.tsx:464]: `shoudMoveContainer`

**Review by akarjakin — APPROVED** [review]

**Pull request merged**